switchbot/adv_parsers/keypad.py

```diff
@@ -0,0 +1,32 @@
+"""Keypad parser."""
+
+from __future__ import annotations
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def process_wokeypad(
+    data: bytes | None, mfr_data: bytes | None
+) -> dict[str, bool | int | None]:
+    """Process woKeypad services data."""
+    if data is None or mfr_data is None:
+        process_wokeypad.lastStatus = -1
+        return {"battery": None, "attemptState": None}
+
+    _LOGGER.debug("mfr_data: %s", mfr_data.hex())
+    if data:
+        _LOGGER.debug("data: %s", data.hex())
+
+    lastStatus = process_wokeypad.lastStatus
```

Comment:
This won't work if they have multiple keypads. Its effectively the same as a global

Reply:
Yeah I just realized that whoops, I'll move it into a device class